@here/harp-mapview/lib/text/TextElement.ts

```diff
@@ -571,6 +449,40 @@ export class TextElement {
         );
     }
 
+    /**
+     * Determine if the `TextElement` is a path label.
+     * @returns `true` if this `TextElement` is a path label.
+     */
+    get isPathLabel(): boolean {
+        return this.path !== undefined && !this.isLineMarker;
+    }
+
+    /**
+     * Determine if the `TextElement` is a poi label.
+     * @returns `true` if this `TextElement` is a poi label.
+     */
+    get isPoiLabel(): boolean {
+        return this.path === undefined;
+    }
+
+    /**
+     * Determine the type of `TextElement`.
+     * @returns This `TextElement` type.
+     */
+    get type(): TextElementType {
+        if (this.isPoiLabel) {
+            return TextElementType.PoiLabel;
+        } else if (this.isLineMarker) {
+            return TextElementType.LineMarker;
+        }
+
+        return TextElementType.PathLabel;
+    }
+
+    get renderState(): TextElementState {
```

Comment:
I think this should be renamed, the fact that you can type

renderState.textRenderState

Where renderState is a TextElementState and not a RenderState and textRenderState is a RenderState, is quite confusing.